daemon/containerd/image_events.go

```diff
@@ -9,8 +9,7 @@ import (
 )
 
 // LogImageEvent generates an event related to an image with only the default attributes.
-func (i *ImageService) LogImageEvent(imageID, refName string, action events.Action) {
-	ctx := context.TODO()
+func (i *ImageService) LogImageEvent(ctx context.Context, imageID, refName string, action events.Action) {
 	attributes := map[string]string{}
 
 	img, err := i.GetImage(ctx, imageID, backend.GetImageOpts{})
```

Comment:
Wonder if this needs some WithoutCancel as it's called on `defer` in various cases

Reply:
I think it depends on the case. The only usage in `defer` I see now is: https://github.com/moby/moby/pull/49014/files#diff-911a739572dfba17ed589736bce4dc155519a814c86a4c8d72a6ae556feab63bR67

Which definitely would need a `WithoutCancel`, but I think it should be done on the caller side.

Reply:
Yeah, makes sense; I first peeked at the code below this line, and it should handle cancelled contexts gracefully (in which case it just skips image-labels in the event), so probably still would be ok.

Looking at that one case where we use `defer`, it looks like it doesn't have to be a defer, so perhaps I'll just  move it to the end, and add a `WithoutCancel()`.

Reply:
OK; updated; https://github.com/moby/moby/compare/d3ed1250aab9a5feeede58a9c24223bb7002985d..ba8f0378f9dbc5306954de22a6311cd0f09acc22

Reply:
OTOH, given our current usages, it would make sense to make `LogImageEvent` non-cancelable by default.
Mostly my thinking is: if we're calling this function, then something **already** happened, so we could be missing an event while the actual effect will not be rolled back.

UNLESS, the calling function would actually be able to roll-back the effects that the event advertises.
However, I don't think it would make much sense to roll back in case the operation gets cancelled just at the time of logging the event.. As it only really happens after everything has already been done.

So, given the above, I think it's also fine to just slap a `ctx = context.WithoutCancel(ctx)` at the top of `LogImageEvent`

Reply:
Yeah, perhaps that makes sense. My slight concern was if the `GetImage(` would take a long time; would that be problematic? (just for the extra metadata) But we already do so before this PR, so I guess this would not make a difference here.